File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if(numRows==1):
            return s

        ans=''
        increment=(numRows-1)*2
        for i in range(numRows):
            for j in range(i,len(s),increment):
                ans+=s[j]
                middleindex=j+increment-2*i
                if (i>0 and i<numRows-1 and middleindex<len(s)):
                    ans+=s[middleindex]
        return ans




        

File: 0006-zigzag-conversion/0006-zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows <= 1 || numRows >= s.size()) {
            return s;
        }
        string ans="";
        int increment=(numRows-1)*2;
        for(int i=0;i<numRows;i++){
            for(int j=i;j<s.size();j=j+increment){
                ans=ans+s[j];
                int diagIndex = j + increment - 2 * i;
                if(i>0&&i<numRows-1&& diagIndex<s.size()){
                    ans=ans+s[diagIndex];
                }
            }
        }
        return ans;
        
    }
};